crates/agent/src/services/agent.rs

```diff
@@ -557,6 +565,160 @@ impl AgentServiceTrait for AgentService {
         self.community_repo.get_by_topic(topic).await
     }
 
+    /// Get proposal arguments with caching and deep research integration
+    async fn get_proposal_arguments(
+        &self,
+        proposal: &Proposal,
+    ) -> Result<CachedResponse<crate::models::analysis::ProposalArguments>> {
+        // Create a cache query based on the proposal content hash
+        let query = CacheableQuery::new("/pre-filter/arguments", "POST").with_body(proposal)?;
+
+        let proposal_clone = proposal.clone();
+        self.cache_service
+            .cache_or_compute(&query, || async {
+                // Extract topic from proposal title or first few lines
+                let topic = extract_topic_from_proposal(&proposal_clone);
+                // Perform deep research on the topic to gather community opinions
+                let deep_research_result = self.compute_deep_research(&topic).await;
+                // Format the deep research data for the AI prompt
+                let community_context = match deep_research_result {
+                    Ok(research) => format_deep_research_for_prompt(&research),
+                    Err(e) => {
+                        warn!("Failed to get deep research for proposal arguments: {}", e);
+                        String::new() // Empty string if research fails
+                    }
+                };
+                // Generate arguments with enhanced prompt that includes deep research
+                let prompt = format!(r#"You are an expert in analyzing governance proposals. Your task is to extract balanced and comprehensive arguments for and against the following proposal.
```

Comment:
Can you put prompt generation logic into separate function / constant?

Reply:
https://github.com/NethermindEth/wei/pull/112/commits/cce63120b954c972e575a254b220131d587f69af